### ISSUE TITLE
Click tree node label?

### ISSUE THREAD

**Issue by EvilPudding** [vurtun/nuklear#626]
Is there any way to use the label of each tree node as a link to open another window? This would be tremendously useful! Thanks in advance.

**Comment by EvilPudding** [comment]
No suggestions about this? Currently I'm using a button inside the tree content, but then I always have to expand before opening it, which is not what I want...

**Comment by vurtun** [comment]
Hey @EvilPudding,
thanks for the feature request. I am not sure what your request is exactly about. Could you clarify for me what exact behavior you want supported from Nuklear.

**Comment by EvilPudding** [comment]
Thanks so much for taking your time!

I am making a game engine, which has a hierarchy system, so each entity may have child entities, each entity also has it's own menu. I am displaying this hierarchy system as a sequence of tabs and nodes as in the image below. I would like to be able to click the tab header to open the specific entity's menu, but because I do not know how to do this, I had to attach a button for each of the tabs, so every time I want the menu of the entity, I have to first expand, and only then click the button.

![nodes](https://user-images.githubusercontent.com/7356051/36696572-7aedd102-1b3c-11e8-9ec1-540a26515f4c.png)

What I would like it to be like in cinema4D:

![nodes2](https://user-images.githubusercontent.com/7356051/36696636-a01c8586-1b3c-11e8-970b-375c1163da51.png)

**Comment by vurtun** [comment]
So if I understand correctly. You want a tree node to be selectable? Or just want to access or control the state of the tree node to see if it was opened or closed?

> I would like to be able to click the tab header to open the specific entity's menu, 

So basically you want to know if a tree node was opened so you can display information in another widget or window? If not could you explain to me what you mean with `entity's menu`.

**Comment by EvilPudding** [comment]
The entity's menu is just a window which is opened when I press the button in my first image, I would like the function of the button to be merged with the tab. I don't want the tab to expand when I click it, I only want the tab to expand if I click the arrow.

A tab's only purpose in this use is just to list the entities' hierarchy, and be able to select a specific one.

**Comment by EvilPudding** [comment]
I understand if it maybe goes against how the tree works right now, because the condition only enters if the tab is expanded, but maybe have a way to attach events?

**Comment by vurtun** [comment]
Oh ok. The behavior should be relatively easy to implement in its own widget:

Something like this:
```c
#define nk_tree_element_push(ctx, type, title, state, selected)
#define nk_tree_element_push_id(ctx, type, title, state, selected, id)
int nk_tree_element_push_hashed(struct nk_context*, enum nk_tree_type, const char *title, enum nk_collapse_states initial_state, int *selected, const char *hash, int len,int seed);
void nk_tree_element_pop(struct nk_context*);
```
and use it like this:

```c
static nk_uint selected = 0;
/* ... */
int was_selected = (selected == nk_hash("Entity", 0));
int is_selected = was_selected;
if (nk_tree_element_push_hashed(ctx, NK_TREE_NODE, "Entity", NK_COLLAPSED, &is_selected, "Entity", strlen("Entity"), 0);
{
    /* go deeper into the tree */
    nk_tree_element_pop(ctx);
}
if (!was_selected && is_selected) {
    /* tree element is now selected */
    selected = nk_hash("Entity", 0);  
} else if (was_selected && !is_selected) {
    /* tree element is now deselected */
    selected = 0;
}    
```
You just have to copy & paste together the code for the tree push implementation and `nk_selectable`.

**Comment by EvilPudding** [comment]
I haven't looked that much into nuklear's source code, and would rather stay out of it, if you had the time to merge the necessary parts I would be very pleased. If not, I understand and will do it eventually.

**Comment by EvilPudding** [comment]
I'm looking into how to do this, and have no idea what I am looking at...

**Comment by vurtun** [comment]
I may have some time on the weekend and can take a look at it.

**Comment by EvilPudding** [comment]
I know I must sound like an jerk, but any progress in this? Sorry in advance.

**Comment by vurtun** [comment]
I can assure you I haven't forgotten you. You are still in my head :smile:. If everything works out today I may have some time tomorrow.

**Comment by vurtun** [comment]
Hey @EvilPudding,
I added selectable tree nodes as well as some code how to use it inside `overview.c`. This is the last time I take requests for widget implementations. This literally was just copy and paste and did not require any specific knowledge of the underlying architecture. Yet this issue was in my head for way to long and caused mental stress that I currently do not need.

**Comment by EvilPudding** [comment]
I guess I'm an idiot and should have looked more into it, sorry and thanks for your work.

**Comment by EvilPudding** [comment]
It works perfect, except line 2941 in the definition of the `push_id` macro, it's missing the state argument, should I pull request? Seems like such a minor fix for a whole commit.

**Comment by vurtun** [comment]
Should hopefully be fixed in commit: d5f32ee6f9aeb17c43a078a2f91c50f2f668beec
